packages/braid-design-system/src/lib/components/private/Typography/Typography.tsx

```diff
@@ -43,7 +43,7 @@ export const Typography = ({
     );
 
   if (process.env.NODE_ENV !== 'production') {
-    if (typeof truncate !== 'undefined') {
+    if (typeof truncate !== 'undefined' && truncate) {
```

Comment:
Can't this just be a truthy check?
```suggestion
    if (truncate) {
```